File: 1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.js
/**
 * @param {number[]} arr
 * @param {number} k
 * @param {number} threshold
 * @return {number}
 */
var numOfSubarrays = function (arr, k, threshold) {
  threshold *= k;
  let res = 0;
  let sum = 0;
  for (let i = 0; i < k; i++) {
    sum += arr[i];
  }
  if (sum >= threshold) {
    res++;
  }
  for (let i = k; i < arr.length; i++) {
    sum += arr[i] - arr[i - k];
    if (sum >= threshold) {
      res++;
    }
  }
  return res;
};

File: 1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.cpp
class Solution
{
public:
  int numOfSubarrays(vector<int> &arr, int k, int threshold)
  {
    threshold *= k;
    int sum = 0;
    int count = 0;
    for (int i = 0; i < k; i++)
    {
      sum += arr[i];
    }
    if (sum >= threshold)
    {
      count++;
    }
    for (int i = k; i < arr.size(); i++)
    {
      sum += arr[i] - arr[i - k];
      if (sum >= threshold)
      {
        count++;
      }
    }
    return count;
  }
};